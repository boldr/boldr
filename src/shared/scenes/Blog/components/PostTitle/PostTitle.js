import React, { PropTypes } from 'react';

const PostTitle = props => {
  let classN = 'boldr-post__title';
  if (props.classname) {
    classN = `${props.classname} boldr-post__title`;
  }

  return <h1 className={ classN }>{props.title}</h1>;
};

export default PostTitle;

PostTitle.propTypes = {
  title: PropTypes.string,
  classname: PropTypes.string,
};
