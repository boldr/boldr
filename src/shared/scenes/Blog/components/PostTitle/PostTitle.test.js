import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import PostTitle from './PostTitle';

describe('<PostTitle />', () => {
  test('renders <PostTitle /> without breaking', () => {
    const wrapper = shallow(<PostTitle title="Random Post Title" />);
    expect(wrapper.find('h1').length).toBe(1);
  });

  it('accepts props and renders them.', () => {
    const wrapper = shallow(<PostTitle title="Random Post Title" />);
    expect(wrapper.instance().props.title).toBe('Random Post Title');
  });
  it('renders snapshot', () => {
    const wrapper = shallow(<PostTitle created="Random Post Title" />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
