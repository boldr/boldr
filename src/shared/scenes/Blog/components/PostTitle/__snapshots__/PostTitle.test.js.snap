// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostTitle /> renders snapshot 1`] = `
<h1
  className="boldr-post__title"
/>
`;
